### ISSUE TITLE
File not closed properly after usage on Windows

### ISSUE THREAD

**Issue by abx78** [JuliaLang/julia#43673]
Even on basic scenarios, a file on windows does not get released correctly by the process 

```
data = open(file_name) do f
    read(f, String)
end
```

After this, the Computer Management on Windows will show the file in the Open Files for a while.

> julia> versioninfo()
> Julia Version 1.6.4
> Commit 35f0c911f4 (2021-11-19 03:54 UTC)
> Platform Info:
>   OS: Windows (x86_64-w64-mingw32)
>   CPU: Intel(R) Xeon(R) Gold 6242 CPU @ 2.80GHz
>   WORD_SIZE: 64
>   LIBM: libopenlibm
>   LLVM: libLLVM-11.0.1 (ORCJIT, skylake-avx512)

**Comment by ararslan** [comment]
@abx78 and I were discussing this a bit in Slack. Interestingly, the call to `CloseHandle` is necessary for Windows to release the file handle but the return code from the call is 0, which [supposedly denotes an error](https://docs.microsoft.com/en-us/windows/win32/api/handleapi/nf-handleapi-closehandle). I wonder whether https://github.com/JuliaLang/julia/blob/master/base/stream.jl#L370 isn't doing the right thing on Windows? I can't figure out where that `_close` actually comes from.

**Comment by IanButterworth** [comment]
Related issue? https://github.com/JuliaLang/julia/issues/39457

**Comment by vtjnash** [comment]
I don't see this. When I run the code above, it appears in Computer Management. Then when I close the file and click the refresh button in Computer Management it disappears, as expected.

**Comment by yummy527** [comment]
OS: windows 10
Julia Version: 1.9.2

When trying to  `rm` a file after open and check the file, I often get the error "resource busy or locked". I suspect that something is wrong with the `open` or `read` function, which leading me here.

By the way, the same code does not trigger an error on my Linux computer.

Please fix it on Windows.

**Comment by vtjnash** [comment]
Yes, posix-compliant kernels don't have this problem; it is unique to the NT kernel implementation. This is often a problem with running virus checkers, so do you happen to have one running?

**Comment by abx78** [comment]
In my case, I did.

**Comment by BioTurboNick** [comment]
FWIW I'm not seeing this on Julia 1.12, Windows 11 25H2
